sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -864,10 +864,18 @@ _pi_queue::_pi_queue(std::vector<ze_command_queue_handle_t> &ComputeQueues,
                      pi_context Context, pi_device Device,
                      bool OwnZeCommandQueue,
                      pi_queue_properties PiQueueProperties,
-                     int ForceComputeIndex)
+                     int ForceComputeIndex, bool OldAPI)
     : Context{Context}, Device{Device}, OwnZeCommandQueue{OwnZeCommandQueue},
       Properties(PiQueueProperties) {
 
+  // Set the type of commandlists the queue will use.
+  bool Default;
+  UsingImmCmdLists = Device->useImmediateCommandLists(Default);
+  if (OldAPI && Default)
+    // The default when called from pre-compiled binaries is to not use
+    // immediate command lists.
+    UsingImmCmdLists = false;
```

Comment:
But any binary app (including existing) running through new SYCL RT will call to **new** piextQueueCreate2. I don't see how we distinguish existing binaries vs re-compiled.

Reply:
My impression was that the SYCL headers map the SYCL queue creation, make_queue and get_native APIs into internal names into the SYCL runtime. Thus, the old headers would call one set of APIs in the runtime and the new headers would call the "2" variants. That way, the headers would decide whether it was old code or recompiled code. 

Unfortunately, this doesn't seem to be working. I am trying to figure out whether this is a fundamental problem or can be solved.

Reply:
make_queue and get_native coming from old/new code can be distinguished.
The problem is SYCL queue()... need a solution for that.